scopes/harmony/application/deploy.task.ts

```diff
@@ -43,21 +43,32 @@ export class DeployTask implements BuildTask {
 
   private async runForOneApp(app: Application, capsule: Capsule, context: BuildContext): Promise<void> {
     const aspectId = this.application.getAppAspect(app.name);
-    let buildTask: TaskResults | undefined;
-    let _metadata;
     if (!aspectId) return;
 
     if (!capsule || !capsule?.component) return;
 
-    if (context?.previousTasksResults.length !== 0) {
-      buildTask = this.getBuildTask(context.previousTasksResults, context.envRuntime.id);
-      _metadata = this.getBuildMetadata(buildTask as TaskResults, capsule.component.id, app);
-    }
+    const buildTask = this.getBuildTask(context.previousTasksResults, context.envRuntime.id);
+
+    const metadata = buildTask
+      ? this.getBuildMetadata(buildTask as TaskResults, capsule.component.id, app)
+      : this.getBuildMetadataFromCompObject(capsule.component, app);
+
+    if (!metadata) return;
```

Comment:
In case there is no metadata you return here
which means the `appDeployContext` will miss the 
```
{
      capsule,
      appComponent: capsule.component,
    }
```
which assigned below

Reply:
right. I guess we don't want to deploy in this case. At least, that's how it was before. We used to check whether the buildTasks are there and if not, return.